playbooks/openstack/resources.py

```diff
@@ -123,11 +130,12 @@ def build_inventory():
     # Use an environment variable to optionally skip returning the app nodes.
     show_compute_nodes = os.environ.get('OPENSTACK_SHOW_COMPUTE_NODES', 'true').lower() == "true"
 
-    # TODO(shadower): filter the servers based on the `OPENSHIFT_CLUSTER`
-    # environment variable.
+    # If `OPENSHIFT_CLUSTER_ID` env variable is defined then it's used to
+    # filter servers by metadata.clusterid attribute value.
```

Comment:
I may be misunderstanding the intent here, but how does OPENSHIFT_CLUSTER_ID get set in the stack's server's metadata?

Reply:
In heat template there is field called cluster_id [1] that combined from openshift_openstack_clusterid and openshift_openstack_public_dns_domain [2][3]


[1] https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_openstack/templates/heat_stack_server.yaml.j2#L23
[2] https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_openstack/templates/heat_stack.yaml.j2#L631
[3] https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_openstack/defaults/main.yml#L36

Reply:
Ah, so is the idea that the user would set OPENSHIFT_CLUSTER_ID in their environment *and* openshift_openstack_clusterid in the inventory?

Reply:
yes and openshift_openstack_clusterid is in the inventory  by default. We couldn't get this option from inventory, that's why env variable is used. If 'OPENSHIFT_CLUSTER_ID' is not defined then it's not used in filtering for servers.

Reply:
*We couldn't get this option from openstack dynamic inventory.py, that's why env variable is used.